Makefile

```diff
@@ -82,55 +84,37 @@ endif
 endif
 
 
-# This to avoid 'Make' options treated as file
-.PHONY: all compile package clean
+# Get the index of "build-docs" rule on command line args
+ifeq "$(filter build-docs,$(MAKECMDGOALS))" "build-docs"
+ifeq "$(MAKE_VERBOSE)" "true"
+	CMD = bash bin/get_argument.sh -v $(MAKECMDGOALS) -s build-docs
+else
+	CMD = bash bin/get_argument.sh $(MAKECMDGOALS) -s build-docs
+endif
+
+BUILD_DOCS_ARG := $(shell $(CMD))
+
+
+# Check if the "build-docs"'s index is not at first argument,
+# then it will returns error
+ifneq "$(BUILD_DOCS_ARG)" "-1"
+ifneq "$(words $(MAKECMDGOALS))" "1"
+$(error $(PREFIX) 'build-docs' rule must be a standalone rule)
+endif
+endif
+
+endif
+
+# This to prevent all `Make`'s rules treated as file by default
+.PHONY: $(ALL_RULES)
 
 
 all:
 	$(info [Makefile-jmatrix])
-	@echo "Options:"
-	@echo "   * compile       - Compile the program."
-	@echo "   * package       - Create archived package (jar) of compiled program."
-	@echo "                        WARNING: Program need to be compiled first!"
-	@echo "   * clean         - Clean all of compiled program and created jar."
-	@echo "   * cleanbin      - Clean all generated class files only."
-	@echo "   * check-verbose - Check the verbose status."
-	@echo "   * usage         - Print the example usages for build the project."
-	@echo ""
-	@echo "Additional Options:"
-	@echo "   * Activating verbose output"
-	@echo ""
-	@echo "       \`export VERBOSE=true\`"
-	@echo ""
-	@echo "     Or:"
-	@echo ""
-	@echo "       \`make [options] VERBOSE=true\`"
-	@echo ""
-	@echo ""
-	@echo "   * Include source files while packaging"
-	@echo ""
-	@echo "       \`make [options] INCLUDE-SRC=true\`"
-	@echo ""
-	@echo ""
-	@echo "   * Invoke the linter"
-	@echo ""
-	@echo "       \`make [options] LINT=true\`"
-	@echo ""
-	@echo ""
-	@echo "   * Add some options or flags to compiler"
-	@echo ""
-	@echo "       \`make [options] FLAGS[=<flags>]\`"
-	@echo ""
-	@echo "Usage:"
-	@echo "     $$ make [options] [...] [arguments]"
-	@echo "     $$ make [options] VERBOSE[=<bool>] INCLUDE-SRC[=<bool>]"
-	@echo "     $$ make [options] (LINT[=<bool>] | FLAGS[=<flags>])"
-	@echo ""
-	@echo "Tips:"
-	@echo "   - Combine the options, Makefile can understand multiple rules."
-	@echo ""
-	@echo "Author:"
-	@echo "   Ryuu Mitsuki"
+	$(if $(shell [ ! -f $(DOCS_PATH)makefile-usage.txt ] && echo 1),\
+		$(error $(PREFIX) File "$(DOCS_PATH)makefile-usage.txt" is missing)\
+	)
+	@cat $(DOCS_PATH)makefile-usage.txt
```

Comment:
The usage instructions and list of available rules and options got moved to `makefile-usage.txt`.
And here, the only way to print the usage instructions it uses `cat` command (Unix built-in).
This also improve the flexibility and less code.